Changes in module 'nf-core/ashlar'
--- modules/nf-core/ashlar/main.nf
+++ modules/nf-core/ashlar/main.nf
@@ -8,9 +8,9 @@
         'biocontainers/ashlar:1.17.0--pyh5e36f6f_0' }"
 
     input:
-    tuple val(meta), path(images)
-    path(opt_dfp)
-    path(opt_ffp)
+    // Stage everything in its own subdirectory to avoid collisions.
+    // dfp and ffp may be empty lists.
+    tuple val(meta), path(images, stageAs: 'image*/*'), path(dfp, stageAs: 'dfp*/*'), path(ffp, stageAs: 'ffp*/*')
 
     output:
     tuple val(meta), path("*.ome.tif"), emit: tif
@@ -20,27 +20,24 @@
     task.ext.when == null || task.ext.when
 
     script:
+    def n_images = images instanceof List ? images.size() : 1
+    def n_dfp = dfp instanceof List ? dfp.size() : 1
+    def n_ffp = ffp instanceof List ? ffp.size() : 1
+    if ( dfp && n_dfp != n_images ) { error 'You must provide the same number of dfp paths as image paths' }
+    if ( ffp && n_ffp != n_images ) { error 'You must provide the same number of ffp paths as image paths' }
+
     def args          = task.ext.args           ?: ''
     def prefix        = task.ext.prefix         ?: "${meta.id}"
-    def dfp           = opt_dfp                 ? "--dfp $opt_dfp" : ""
-    def ffp           = opt_ffp                 ? "--ffp $opt_ffp" : ""
-    def num_files     = images instanceof List  ? images.size()    : 1
-    def opt_dfp_size  = opt_dfp instanceof List ? opt_dfp.size()   : 1
-    def opt_ffp_size  = opt_ffp instanceof List ? opt_ffp.size()   : 1
-    def dfp_validated = (opt_dfp_size == 0 || opt_dfp_size == 1 || opt_dfp_size == num_files) ? true : false
-    def ffp_validated = (opt_ffp_size == 0 || opt_ffp_size == 1 || opt_ffp_size == num_files) ? true : false
-
-    if ( !dfp_validated ) { error "Please input only zero, one, or N dfp files, where N is the number of input images" }
-    if ( !ffp_validated ) { error "Please input only zero, one, or N ffp files, where N is the number of input images" }
-
+    def dfp_arg       = dfp                     ? "--dfp $dfp" : ""
+    def ffp_arg       = ffp                     ? "--ffp $ffp" : ""
     """
 
     ashlar \\
         -o ${prefix}.ome.tif \\
         $images \\
         $args \\
-        $dfp \\
-        $ffp
+        $dfp_arg \\
+        $ffp_arg
 
     sed -i -E 's/UUID="urn:uuid:[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}"/                                                    /g' ${prefix}.ome.tif
 

************************************************************
