Changes in module 'nf-core/ashlar'
--- modules/nf-core/ashlar/main.nf
+++ modules/nf-core/ashlar/main.nf
@@ -8,9 +8,9 @@
         'biocontainers/ashlar:1.17.0--pyh5e36f6f_0' }"
 
     input:
-    tuple val(meta), path(images)
-    path(opt_dfp)
-    path(opt_ffp)
+    tuple val(meta), path(images, stageAs: 'image*/*')
+    path(opt_dfp, stageAs: 'dfp*/*')
+    path(opt_ffp, stageAs: 'ffp*/*')
 
     output:
     tuple val(meta), path("*.ome.tif"), emit: tif

************************************************************
