Changes in module 'nf-core/basicpy'
--- modules/nf-core/basicpy/main.nf
+++ modules/nf-core/basicpy/main.nf
@@ -8,8 +8,8 @@
     tuple val(meta), path(image)
 
     output:
-    tuple val(meta), path("*.tiff"), emit: fields
-    path "versions.yml"            , emit: versions
+    tuple val(meta), path("*-dfp.tiff"), path("*-ffp.tiff"), emit: profiles
+    path "versions.yml",                                     emit: versions
 
     when:
     task.ext.when == null || task.ext.when

************************************************************
